From e8468d8e347130d6a053fa66a4b55efb0ca7a28a Mon Sep 17 00:00:00 2001
From: fafryd <dz1125.bug.tracker@gmail.com>
Date: Wed, 25 Sep 2019 20:38:39 +0200
Subject: [PATCH] Revert "meson: re-add incorrect pkg-config files with GLVND
 for backward compatibility"

This reverts commit e59e9cd58a95d8dc9b06d09f2abed96ba7b1a1b9.
---
 meson.build          |  4 ----
 src/egl/meson.build  | 27 ++++++++++-----------------
 src/mapi/meson.build |  2 +-
 src/meson.build      | 14 ++------------
 4 files changed, 13 insertions(+), 34 deletions(-)

diff --git a/meson.build b/meson.build
index 4afd8ca663b..52f354abfb1 100644
--- a/meson.build
+++ b/meson.build
@@ -1309,10 +1309,6 @@ endif
 dep_glvnd = null_dep
 if with_glvnd
   dep_glvnd = dependency('libglvnd', version : '>= 0.2.0')
-  # GLVND until commit 0dfaea2bcb7cdcc785f9 ("Add pkg-config files for EGL, GL,
-  # GLES, and GLX.") was missing its pkg-config files, forcing every vendor to
-  # provide them and the distro maintainers to resolve the conflict.
-  glvnd_missing_pc_files = dep_glvnd.version().version_compare('< 1.2.0')
   pre_args += '-DUSE_LIBGLVND=1'
 endif
 
diff --git a/src/egl/meson.build b/src/egl/meson.build
index 7038a68e955..14aca2a2186 100644
--- a/src/egl/meson.build
+++ b/src/egl/meson.build
@@ -173,25 +173,18 @@ libegl = shared_library(
   version : egl_lib_version,
 )
 
-# If using glvnd the pkg-config header should not point to EGL_mesa, it should
-# point to EGL. glvnd is only available on unix like platforms so adding -l
-# should be safe here
-if with_glvnd and glvnd_missing_pc_files
-  _egl = '-L${libdir} -lEGL'
-else
-  _egl = libegl
+if not with_glvnd
+  pkg.generate(
+    name : 'egl',
+    description : 'Mesa EGL Library',
+    version : meson.project_version(),
+    libraries : libegl,
+    libraries_private: gl_priv_libs,
+    requires_private : gl_priv_reqs,
+    extra_cflags : gl_pkgconfig_c_flags,
+  )
 endif
 
-pkg.generate(
-  name : 'egl',
-  description : 'Mesa EGL Library',
-  version : meson.project_version(),
-  libraries : _egl,
-  libraries_private: gl_priv_libs,
-  requires_private : gl_priv_reqs,
-  extra_cflags : gl_pkgconfig_c_flags,
-)
-
 if with_tests and prog_nm.found()
   if with_glvnd
     egl_symbols = files('egl-glvnd-symbols.txt')
diff --git a/src/mapi/meson.build b/src/mapi/meson.build
index 39c1dba7ce0..2c79a04f1df 100644
--- a/src/mapi/meson.build
+++ b/src/mapi/meson.build
@@ -35,7 +35,7 @@ if with_shared_glapi
 else
   libglapi = []
 endif
-if not with_glvnd or glvnd_missing_pc_files
+if not with_glvnd
   if with_gles1
     subdir('es1api')
   endif
diff --git a/src/meson.build b/src/meson.build
index bc677d890cf..69ad05ea03b 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -106,22 +106,12 @@ endif
 
 # This must be after at least mesa, glx, and gallium, since libgl will be
 # defined in one of those subdirs depending on the glx provider.
-if with_glx != 'disabled'
-  # If using glvnd the pkg-config header should not point to GL_mesa, it should
-  # point to GL. glvnd is only available on unix like platforms so adding -l
-  # should be safe here
-  # TODO: in the glvnd case glvnd itself should really be providing this.
-  if with_glvnd and glvnd_missing_pc_files
-    _gl = '-L${libdir} -lGL'
-  else
-    _gl = libgl
-  endif
-
+if with_glx != 'disabled' and not with_glvnd
   pkg.generate(
     name : 'gl',
     description : 'Mesa OpenGL Library',
     version : meson.project_version(),
-    libraries : _gl,
+    libraries : libgl,
     libraries_private : gl_priv_libs,
     requires_private : gl_priv_reqs,
     variables : ['glx_tls=yes'],
-- 
2.23.0

